rllib/utils/metrics/stats.py

```diff
@@ -541,111 +558,112 @@ def __repr__(self) -> str:
-            )
+            self._comp_error("__gt__")
         else:
             return float(self) > float(other)
 
     def __add__(self, other):
         if self._reduce_method is None:
-            raise ValueError(
-                "Cannot add Stats object with reduce method `None` to other because it can not be reduced to a single value."
-            )
+            self._comp_error("__add__")
         else:
             return float(self) + float(other)
 
     def __sub__(self, other):
         if self._reduce_method is None:
-            raise ValueError(
-                "Cannot subtract Stats object with reduce method `None` from other because it can not be reduced to a single value."
-            )
+            self._comp_error("__sub__")
         else:
             return float(self) - float(other)
 
     def __mul__(self, other):
         if self._reduce_method is None:
-            raise ValueError(
-                "Cannot multiply Stats object with reduce method `None` with other because it can not be reduced to a single value."
-            )
+            self._comp_error("__mul__")
         else:
             return float(self) * float(other)
 
     def __format__(self, fmt):
         if self._reduce_method is None:
             raise ValueError(
-                "Cannot format Stats object with reduce method `None` because it can not be reduced to a single value."
+                "Cannot format Stats object with reduce method `None` because it can "
+                "not be reduced to a single value."
             )
         else:
             return f"{float(self):{fmt}}"
 
+    def _comp_error(self, comp):
+        raise ValueError(
+            f"Cannot {comp} Stats object with reduce method `None` to other "
+            "because it can not be reduced to a single value."
+        )
+
     def get_state(self) -> Dict[str, Any]:
         state = {
-            "values": self.values,
+            # Make sure we don't return any tensors here.
+            "values": convert_to_numpy(self.values),
             "reduce": self._reduce_method,
             "reduce_per_index_on_aggregate": self._reduce_per_index_on_aggregate,
             "window": self._window,
             "ema_coeff": self._ema_coeff,
             "clear_on_reduce": self._clear_on_reduce,
             "_hist": list(self.get_reduce_history()),
+            "_could_be_tensor": self._could_be_tensor,
         }
         if self._throughput_stats is not None:
             state["throughput_stats"] = self._throughput_stats.get_state()
         return state
 
     @staticmethod
-    def from_state(state: Dict[str, Any], throughputs=False) -> "Stats":
+    def from_state(state: Dict[str, Any]) -> "Stats":
+        # If `values` could contain tensors, don't reinstate them (b/c we don't know
+        # whether we are on a supported device).
+        values = state["values"]
+        if "_could_be_tensor" in state and state["_could_be_tensor"]:
+            values = []
```

Comment:
1. If we do this, we should state it as a known limitation in metrics-logger.rst.
2. Alternatively, can we noch check whether we are on a supported device and keep track of GPU tensors?